src/libsyntax/parse/parser/diagnostics.rs

```diff
@@ -902,18 +876,61 @@ impl<'a> Parser<'a> {
             }
         }
         let sm = self.sess.source_map();
-        match (sm.lookup_line(prev_sp.lo()), sm.lookup_line(sp.lo())) {
-            (Ok(ref a), Ok(ref b)) if a.line == b.line => {
-                // When the spans are in the same line, it means that the only content
-                // between them is whitespace, point only at the found token.
-                err.span_label(sp, label_exp);
+        if !sm.is_multiline(prev_sp.until(sp)) {
```

Comment:
The logic here seems very similar to the case in my comment above -- a refactoring opportunity?